components/compliance-service/ingest/pipeline/processor/reports_project_tagger.go

```diff
@@ -29,16 +33,32 @@ func reportProjectTagger(in <-chan message.Compliance, authzClient iam_v2.Projec
 	}
 	out := make(chan message.Compliance, 100)
 	go func() {
+		nextNumToDrop := 1
 		bundleSize := 0
 		var projectRulesCollection map[string]*iam_v2.ProjectRules
 		for msg := range in {
+			if msg.Ctx.Err() != nil {
+				msg.FinishProcessingCompliance(msg.Ctx.Err())
+				continue
+			}
 			if bundleSize <= 0 {
-				bundleSize = len(in)
+				nextBundleSize := len(in)
 				logrus.WithFields(logrus.Fields{
 					"message_id": msg.Report.ReportUuid,
-					"bundleSize": bundleSize,
+					"bundleSize": nextBundleSize,
 				}).Debug("BundleProjectTagging - Update Project rules")
-				projectRulesCollection = getProjectRulesFromAuthz(msg.Ctx, authzClient)
+				var err error
+				projectRulesCollection, err = getProjectRulesFromAuthz(authzClient)
+				if err != nil {
+					dropComplianceMessages(in, err, nextNumToDrop-1)
```

Comment:
This section keeps falling more and more messages until all the authz request works. I don't understand why the number of requests dropped increases. Why don't we drop all the messages that are currently in the queue?

Reply:
We could also loop through all the messages and drop only the ones that have a context that has timed out. Then try the auth request again.

Reply:
the idea is that we don't want to drop the entire queue if we see a blip from authz. Instead, we will ease into dropping large amounts of messages. So the first one, we only drop the message that is being processed, then we drop ourselves + another, then ourselves + 2*other, then ourself + 2^n*other.

Reply:
I think the first dropped message will not call `m.FinishProcessingCompliance(err)`, because dropComplianceMessages gets called with numToDrop = 0.

Reply:
So, you are increasing the number of dropped messages for every time the authz call fails, but why not increase the number of messages drop based on the load of the system? So, if there is one message coming in every 5 seconds then this would drop just one message at a time. If there were 100 messages every 5 seconds it would start to get backed up and then exponentially start dropping more and more. The problem is at first, the queue might fill up when there is a heavy load.

Reply:
good catch, i'll add the finish processing call.

As far as shedding load, there are a number of things that can be done. The reason I don't want to drop the entire queue is because i'm not convinced there isn't a problem with making calls to authz. Retrying once may get us back on track. There's another PR out that tries to deal with queues filling up that I'm working on.